src/aggregator/integration/placement_change_test.go

```diff
@@ -0,0 +1,303 @@
+		require.NoError(t, server.startServer())
+		log.Sugar().Infof("server %d is now up", i)
+	}
+
+	// Create clients for writing to the servers.
+	clients := make([]*client, 0, len(servers))
+	for _, server := range servers {
+		client := server.newClient()
+		require.NoError(t, client.connect())
+		clients = append(clients, client)
+	}
+
+	for _, server := range servers {
+		require.NoError(t, server.waitUntilLeader())
+	}
+
+	var (
+		idPrefix = "metric.id"
+		numIDs   = 100
+
+		start1    = getNowFn()
+		stop1     = start1.Add(10 * time.Second)
+		start2    = stop1.Add(time.Minute + 2*time.Second)
+		stop2     = start2.Add(10 * time.Second)
+		finalTime = stop2.Add(time.Minute + 2*time.Second)
+		interval  = 2 * time.Second
+
+		sleepDuration = time.Second
+	)
+	ids := generateTestIDs(idPrefix, numIDs)
+	testTimedMetadataTemplate := metadata.TimedMetadata{
+		AggregationID: maggregation.MustCompressTypes(maggregation.Sum),
+		StoragePolicy: policy.NewStoragePolicy(2*time.Second, xtime.Second, time.Hour),
+	}
+	metadataFn := func(idx int) metadataUnion {
+		timedMetadata := testTimedMetadataTemplate
+		return metadataUnion{
+			mType:         timedMetadataType,
+			timedMetadata: timedMetadata,
+		}
+	}
+	datasets := []testDataset{
+		mustGenerateTestDataset(t, datasetGenOpts{
+			start:        start1,
+			stop:         stop1,
+			interval:     interval,
+			ids:          ids,
+			category:     timedMetric,
+			typeFn:       roundRobinMetricTypeFn,
+			valueGenOpts: defaultValueGenOpts,
+			metadataFn:   metadataFn,
+		}),
+		mustGenerateTestDataset(t, datasetGenOpts{
+			start:        start2,
+			stop:         stop2,
+			interval:     interval,
+			ids:          ids,
+			category:     timedMetric,
+			typeFn:       roundRobinMetricTypeFn,
+			valueGenOpts: defaultValueGenOpts,
+			metadataFn:   metadataFn,
+		}),
+	}
+
+	for _, data := range datasets[0] {
+		setNowFn(data.timestamp)
+
+		for _, mm := range data.metricWithMetadatas {
+			idx := getServerIndex(mm.metric.ID(), initialPlacement)
+			require.NoError(t, clients[idx].writeTimedMetricWithMetadata(mm.metric.timed, mm.metadata.timedMetadata))
+		}
+		for _, c := range clients {
+			require.NoError(t, c.flush())
+		}
+
+		// Give server some time to process the incoming packets.
+		time.Sleep(sleepDuration)
+	}
+
+	setNowFn(start2)
+	time.Sleep(sleepDuration)
+	require.NoError(t, setPlacement(placementKey, kvStore, finalPlacement))
+	time.Sleep(sleepDuration)
+
+	for _, data := range datasets[1] {
+		setNowFn(data.timestamp)
+
+		for _, mm := range data.metricWithMetadatas {
+			idx := getServerIndex(mm.metric.ID(), finalPlacement)
+			require.NoError(t, clients[idx].writeTimedMetricWithMetadata(mm.metric.timed, mm.metadata.timedMetadata))
+		}
+		for _, c := range clients {
+			require.NoError(t, c.flush())
+		}
+
+		// Give server some time to process the incoming packets.
+		time.Sleep(sleepDuration)
+	}
+
+	setNowFn(finalTime)
+	time.Sleep(sleepDuration)
```

Comment:
Why do we need this (before stopping the servers)?

Reply:
IIUC, it's waiting for the next tick so it could observe the updated time https://github.com/m3db/m3/pull/3465/commits/99067f0f2cedf940fc74ef7e915fa52f7cd05388